### ISSUE TITLE
CC Tech Roadmap

### ISSUE THREAD

**Issue by mattl** [cc-archive/techstuff#17]
Here's the roadmap, please take a look: https://github.com/creativecommons/techstuff/wiki/Tech-Product-Roadmap-2015

**Comment by tamielbr** [comment]
May I suggest you include other platforms on your list? WP is a great choice, but others such as Omeka have an existing plugin that needs refreshing (https://github.com/cc-archive/jswidget/issues/1).

**Comment by mattl** [comment]
Sounds reasonable. I think we can look at refreshing the JS Widget. @robmyers WDYT?

**Comment by mattl** [comment]
For other platforms in general, see https://github.com/creativecommons/platform-initiative

**Comment by rheaplex** [comment]
@mattl Sure, we can look at JS. Where shall we slot that in? (And we should respond on-list as well...)

**Comment by benadida** [comment]
Looks like a lot of work and great upgrades!

Question: what are the motivations / overarching goals? Would be good to understand them, and then to see how each task fits in. Otherwise it's hard to provide feedback on prioritization / importance of various ideas.

**Comment by jessicacoates** [comment]
I'd like to see a consultation process built in to solicit ideas for the website redesign - both at the front page and later stages. It should go to affiliates at a minimum, even the general public if you like. A lot of the affiliates base their website on ours, and so will have an interest in how it ends up. Plus they're our biggest single user base and so are likely to have good ideas/comments on what needs fixing.

**Comment by mattl** [comment]
@benadida Some of it is just updating and repairing things which are existing but neglected, most of the OER work relates to a grant, which is how we got @robmyers ;) and the rest is stuff we're interested in doing to help build the commons -- analytics, better chooser, etc

**Comment by mattl** [comment]
@jessicacoates We can talk more about this in email, but for the affiliates we're actually building a template that can be used that is distinct from the CC website.

**Comment by mzeinstra** [comment]
I was in the understanding that CC Global would do work on rightsstatements.org I do not see this in your roadmap yet.

Also I agree with @benadida that I miss the overarching goals of the products. What drives your technology and design choices?

**Comment by mattl** [comment]
@mzeinstra It's not clear to me what will be happening with that. We've been busy and the group has been working without us. I suspect we'll be doing some support work.

See above for our goals.

**Comment by mzeinstra** [comment]
@mattl I believe the group still expects that CC delivers the website and infrastructure to publish these statements as you committed to in Amsterdam.

I'm looking for a more overall mission where to place the work of CC Global from a technology perspective and why CC Global believes that this is the way to go. The current document, to me is not a roadmap but a development todo list. Without much technical knowledge you would have no idea what you are doing. A grouping of different tasks into an argument why you want to change such and such technology would give us a much better grasp on your team's work.

**Comment by mattl** [comment]
I'll follow up with Diane on the rightsstatements.org stuff and get it into the roadmap, or todo list, if you prefer ;)

**Comment by nasirkhan** [comment]
Should not we have any plan related to translation/localization ?

**Comment by rheaplex** [comment]
Translation/localization is a core part of each project. That said we do need to ensure that we have the technologies and support in place to make sure that it can be a _smoothly running_ core part of each project.

**Comment by RCEurope1** [comment]
@robmyers well said, but right now the tech we are running is still in the position of unsmoothing all the rest. What @jessicacoates and @mzeinstra are pointing out is at the @root of all this, b/c tech shift is destined to produce serial hick-ups or even fail if it lacks proper stakeholder consultation beforehand. So far, nobody seems to be really sure about how tech decisions are made by whom an when, and those working on the tech seem often enough to be unaware of the consequences. So we need a consultation workflow and I'm pretty sure that – in line with general GitHubization at CC – there are good examples from tech-heavy projects how to do this ...

**Comment by mattl** [comment]
@RCEurope1 I don't agree that we're unaware of the consequences, but I'm not sure what's unclear about this. We're the technology team, we've defined the projects we're going to work on, and when.

**Comment by rheaplex** [comment]
Informally, the technology team's priorities are:
1. Keep the lights on.
2. Satisfy funding requirements.
3. Pursue strategic goals. 

Where my strategic goals include:
1. Demonstrate the value of CC licenses.
2. Help people find CC licensed resources.
3. Grow the commons.

Tech decisions are made by the technology team in order to satisfy these priorities in consultation with stakeholders and with the CEO.

In order to make these decisions we need stakeholders to identify themselves and to work with us to help ensure that requirements are clear, specific and explicit ahead of time.

To help make this happen we've initiated this process.

_Please_ add tasks or implications we've missed to the roadmap document or open a ticket. We want and need your help and input.

We _have_ to update CC's tech infrastructure and move some systems to more appropriate platforms, and we _have_ to prioritize this in such a way that causes as little disruption as possible but also with as little wasted effort as possible. Otherwise we won't be able to do everything we need to do.

In the case of the Wiki we explicitly communicated that the update would be taking place, the scale and complexity of the work, and that some breakage was unavoidable, regrettable, and fixable. Breakage in this instance was not due to lack of stakeholder consultation, it was due to the inherent complexity of making a more than decade-old website in poor repair fit for purpose again. We've already fixed issues reported by stakeholders who have communicated with us via GitHub.

Give us lots of lovely GitHub Issues... :-)

**Comment by RCEurope1** [comment]
@mattl there have been several instances where it at least _looked_ like not all cosequences were considered/known/expected. That is not to say that this is a big fault of the tech team, it's IMHO rather an overarching issue of process.

@robmyers I'd see it the other way round: When the core CC tech systems are touched (tech-wise and regarding message), that means touching something that also belongs to the community of the global network. Taking this distributed ownership view, the affiliates as stakeholders do not have to actively identify themselves – how could they know that some tech change is planned? – but rather have to be notified. At this point the regional coordinators are required to be alert and initiate notice and discussion, but also the tech team needs to be aware that there is this world-wide stakeholder group in the background. And this needs to take into account the structure of the global network, which is why I think that it doesn't suffice to have a tech roadmap. It features 68 bullet points for 2015 alone (yes, I counted them ;) and the network cannot discuss on that wide range at once or even pick out the bullets that require discussion.

And regarding GitHub in general: I see it as a tool for communication, but it doesn't substitute for a consultation and decision making process. The decision to fix a GitHub issue, postpone it or put it to "won't be fixed" is what it is about, not the means of communicating issues. Plus, the global network's standard means of communication is still the mailing list(s), not GitHub. We might be able to change that, but that takes time.

Bottom line from my point of view is that CEO and tech team can indeed keep the decision making between themselves for some things, but less so for anything that draws the global network into something like a joint liability. That can concern tech only questions like what translation system to use as well as changes in message and/or focus of CC as a whole. For that we need a consistent process that at least consists of a consultation, and so far we don't seem to have one really.

**Comment by ryanmerkley** [comment]
There are a lot of good insights here, in particular from @RCEurope1 @mzeinstra and @benadida. I don't want to get into a discussion about who "owns" what -- I'd rather focus on how we want to work together on our shared goals, and making sure those goals are shared means talking to each other more, not less and listening more, not less. 

The roadmap needs a pre-amble that identifies CC's strategic priorities, and makes it clear that this isn't just a to do list. I'll work on that with @mattl and @robmyers, and ping this group when it's updated. I'd also agree it needs, for lack of a better phrase, rules of engagement to ensure that stakeholders are engaged in both the "what are our priorities" phase and also the "what is the work and how will we do it" phase. We'll come back with that shortly.

**Comment by jessicacoates** [comment]
The 4.0 consultation, while intensive, was an excellent example of affiliate consultation that worked well and got affiliates engaged and working within HQ projects. It involved a public wiki with updates,  surveys to get general feedback, regular emails to the affiliates and licences (ie public) lists informing people when things were being added/changed/decided, calls timed for different regions and direct discussion with the most engaged people. Legal coordinated the content and made decisions, the RCs helped with pushing things out and organising calls. I would advocate something that full on (it took 2+ years) for ordinary projects, but the basic format and approach provides a good example for future consultation.
